src/server/pps/server/s3g_sidecar.go

```diff
@@ -58,17 +59,19 @@ func (a *apiServer) ServeSidecarS3G() {
 		retryCtx, retryCancel := context.WithCancel(context.Background())
 		defer retryCancel()
 
-		if err := a.pipelines.ReadOnly(retryCtx).Get(pipelineName, s.pipelineInfo); err != nil {
+		if err := a.txnEnv.WithReadContext(retryCtx, func(txnCtx *txncontext.TransactionContext) error {
+			specCommit, err := a.findPipelineSpecCommit(txnCtx, pipelineName, "")
+			if err != nil {
+				return err
+			}
+			return a.pipelines.ReadWrite(txnCtx.SqlTx).Get(specCommit, s.pipelineInfo)
+		}); err != nil {
 			return errors.Wrapf(err, "sidecar s3 gateway: could not find pipeline")
 		}
 
 		// Set auth token for s.pachClient (pipelinePtr.AuthToken will be empty if
 		// auth is off)
 		s.pachClient.SetAuthToken(s.pipelineInfo.AuthToken)
```

Comment:
do we still need to set the AuthToken here?

Reply:
I think you're a better person to answer that. We don't need it in this function, but I assumed the s3g sidecar would rely on auth elsewhere.

Reply:
I'm not immediately sure whether s3g will end up using this value. I need to look into it. In the meantime leaving it is fine